### ISSUE TITLE
Integration with calico, reuse of calico bgp functions

### ISSUE THREAD

**Issue by duanjiong** [kubesphere/porter#151]
calico can also announce the service ip, where the porter can act as an IPAM to assign addresses.
https://docs.projectcalico.org/networking/advertise-service-ips

**Comment by Elegant996** [comment]
With the imminent release of calico 3.18, calico can also announce LoadBalancer IPs. If one so chooses, they should be able to define an EIP resource and not require the BGP resources (handled by calico).

The is relegates porter to only assigning IPAM addresses but should be a viable option for cluster configuration.

**Comment by duanjiong** [comment]
> With the imminent release of calico 3.18, calico can also announce LoadBalancer IPs. If one so chooses, they should be able to define an EIP resource for porter and not require the BGP resources (handled by calico).
> 
> The relegates porter to only assigning IPAM addresses but should be a viable option for cluster configuration. MetalLB already supports this by removing the speaker daemonset and only have the controller aspect deployed.

Yes, you are right, this is also easy to modify in the porter, we will also modify immediately.
